sources/engine/Stride.Engine/Engine/Splines/Processors/SplineTransformProcessor.cs

```diff
@@ -0,0 +1,122 @@
+//// Copyright (c) Stride contributors (https://Stride.com)
+//// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using Stride.Core.Mathematics;
+using Stride.Engine.Splines.Components;
+using Stride.Engine.Splines.HierarchyTransformOperations;
+using Stride.Graphics;
+using Stride.Rendering;
+
+namespace Stride.Engine.Splines.Processors
+{
+    /// <summary>
+    /// The processor for <see cref="SplineComponent"/>.
+    /// </summary>
+    public class SplineTransformProcessor : EntityProcessor<SplineComponent, SplineTransformProcessor.SplineTransformationInfo>
+    {
+        private HashSet<SplineComponent> splineComponentsToUpdate = new();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SplineTransformProcessor"/> class.
+        /// </summary>
+        public SplineTransformProcessor()
+            : base(typeof(TransformComponent))
+        {
+        }
+
+        protected override SplineTransformationInfo GenerateComponentData(Entity entity, SplineComponent component)
+        {
+            return new SplineTransformationInfo
+            {
+                TransformOperation = new SplineViewHierarchyTransformOperation(component),
+            };
+        }
+
+        protected override bool IsAssociatedDataValid(Entity entity, SplineComponent component, SplineTransformationInfo associatedData)
+        {
+            return component == associatedData.TransformOperation.SplineComponent;
+        }
+
+        protected override void OnEntityComponentAdding(Entity entity, SplineComponent component, SplineTransformationInfo data)
+        {
+            component.Spline.OnSplineDirty += () => splineComponentsToUpdate.Add(component);
+
+            splineComponentsToUpdate.Add(component);
+
+            entity.Transform.PostOperations.Add(data.TransformOperation);
+        }
+
+        protected override void OnEntityComponentRemoved(Entity entity, SplineComponent component, SplineTransformationInfo data)
+        {
+            component.Spline.OnSplineDirty -= () => splineComponentsToUpdate.Add(component);
```

Comment:
Note that this won't work, because you're trying to remove a lambda function which is a different object from the one in the OnAdding method. I would suggest adding an `Update()` method to the `SplineTransformationInfo` which would perform the action within lambda. But passing `data.Update` as a delegate should be reference preserving and properly deregister the event.

Reply:
I you have the time, I could use your help here. I tried moving the action code to a update method and starting playing around with passing all types of delegates, but none of these seem to trigger the removal of the actual component.

Reply:
Hmm, the main thing is that when you're removing the delegate it will look at two things - object instance and pointer to the method. For static methods object instance is null.
When you create a lambda which captures an object you're allocating a new closure each time you see the lambda in code. For static lambdas these are hoisted to a static method and the removal from an event would actually work.

So how do you go about fixing it? The simplest thing you can do is cache the created closure in the context for your component processor.

```csharp
// OnComponentAdding
Action onSplineDirty = () => update...;
data.OnSplineDirtyCallback = onSplineDirty;
component.Spline.OnSplineDirty += onSplineDirty;

// OnComponentRemoved
component.Spline.OnSplineDirty -= data.OnSplineDirtyCallback;
```

Reply:
What I was suggesting earlier was to encapsulate the logic in `data.Update()` and avoid using a lambda expression entirely.